### PR TITLE
Add Approval Confirmation Screen component to Storybook

### PR DESCRIPTION
**Pull request by EtDu**
- Adds Approval Confirmation Screen component to Storybook
- Uses mocked background instance & meta metrics provider for Storybook
- Adds react `Router` with mocked browser history for Storybook
- Adds knob functionality to change origin and icon URL for confirm-approve component

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -8,7 +8,12 @@ import '../ui/css/index.scss';
 import localeList from '../app/_locales/index.json';
 import * as allLocales from './locales';
 import { I18nProvider, LegacyI18nProvider } from './i18n';
+import MetaMetricsContext from './metametrics'
 import testData from './test-data.js';
+import { Router } from "react-router-dom";
+import { createBrowserHistory } from "history";
+import { _setBackgroundConnection } from '../ui/store/actions'
+import { Promise } from 'globalthis/implementation';
```
Comment (kumavis) [inline .storybook/preview.js:16]: why

File: .storybook/metametrics.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        <NewMetaMetricsProvider>
+          <NewLegacyMetaMetricsProvider>
+            {props.children}
+          </NewLegacyMetaMetricsProvider>
+        </NewMetaMetricsProvider>
+      </LegacyMetaMetricsProvider>
+    </MetaMetricsProvider>
+  );
+
+export default MetaMetricsContext
```
Comment (kumavis) [inline .storybook/metametrics.js:24]: `MetaMetricsContext` -> `MetaMetricsMetaProvider` 😆

File: package.json [hunk 3]
```diff
@@ -143,6 +144,7 @@
     "fast-safe-stringify": "^2.0.7",
     "fuse.js": "^3.2.0",
     "globalthis": "^1.0.1",
+    "history": "^5.0.0",
```
Comment (kumavis) [inline package.json:147]: should be devdep

File: package.json [hunk 4]
```diff
@@ -98,6 +98,7 @@
     "@metamask/jazzicon": "^2.0.0",
     "@metamask/logo": "^2.5.0",
     "@metamask/obs-store": "^5.0.0",
+    "@metamask/post-message-stream": "^4.0.0",
```
Comment (kumavis) [inline package.json:101]: why did you need to add this

Comment (EtDu) [reply package.json:101]: I didn't, seems it was auto genned

File: .storybook/preview.js (hunk 1 above)
Comment (EtDu) [reply .storybook/preview.js:16]: Auto added by vscode for some reason

File: ui/pages/confirm-approve/confirm-approve.stories.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import { store } from '../../../.storybook/preview';
+import ConfirmApprove from '.';
+
+export default {
+  title: 'Confirmation Screens',
+};
+
+const PageSet = ({ children }) => {
+  const params = useParams();
+  params.id = 1906703652727041;
```
Comment (kumavis) [inline ui/pages/confirm-approve/confirm-approve.stories.js:16]: what is this magical number

File: ui/pages/confirm-approve/confirm-approve.stories.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  return children;
+};
+
+export const ApproveTokens = () => {
+  // useEffect for creating Redux knobs
+  useEffect(() => {
```
Comment (kumavis) [inline ui/pages/confirm-approve/confirm-approve.stories.js:22]: remove if not needed

File: .storybook/metametrics.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import {
+  MetaMetricsProvider,
+  LegacyMetaMetricsProvider,
+} from '../ui/contexts/metametrics';
+import {
+  MetaMetricsProvider as NewMetaMetricsProvider,
+  LegacyMetaMetricsProvider as NewLegacyMetaMetricsProvider,
+} from '../ui/contexts/metametrics.new';
+
+const MetaMetricsProviderStorybook = (props) =>
```
Comment (kumavis) [inline .storybook/metametrics.js:11]: 👍

**Review by kumavis — COMMENTED** [review]
neat

**Review by brad-decker — DISMISSED** [review]
This is looking dope! @EtDu 🎉 

one question: 
image

is the missing icons a byproduct or the PR build process, or perhaps a pre-existing issue? Non blocking for this to land!

**Comment by EtDu** [comment]
> This is looking dope! @EtDu 🎉
> 
> one question:
> image
> 
> is the missing icons a byproduct or the PR build process, or perhaps a pre-existing issue? Non blocking for this to land!

It must be a CI issue. I'm getting 404s in the dev console for some svgs, but they all appear to be present in the artifacts https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/9625/workflows/9275ccef-99bd-4b29-9668-5ca5977c4486/jobs/280732/artifacts 🤔

**Comment by EtDu** [comment]
Also the Euclid font isn't rendering properly

**Review by kumavis — DISMISSED** [review]

**Comment by kumavis** [comment]
partially fixed fonts here but couldn't solve fontawesome https://github.com/MetaMask/metamask-extension/pull/11163

**Review by brad-decker — APPROVED** [review]
